pkg/bundle/reader.go

```diff
@@ -0,0 +1,118 @@
+package bundle
+
+import (
+	"archive/tar"
+	"fmt"
+	"io"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/tektoncd/pipeline/pkg/client/clientset/versioned/scheme"
+	tkremote "github.com/tektoncd/pipeline/pkg/remote/oci"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// ObjectVisitor is an input function that callers of this file's methods can implement to act on the read contents of a
+// Tekton bundle.
+type ObjectVisitor func(gvk schema.GroupVersionKind, name string, element runtime.Object, raw []byte)
```

Comment:
Thinking outloud here : this part might be useful in `tektoncd/pipeline` right ? (for other component, tooling to re-use, even e2e tests, …)

Reply:
Sure. I don't have a particular location in mind but if you have on, I can push into there instead

Reply:
Not sure yet too, we can do this later one though, no need to make this PR depend on some PR in pipeline